src/Agda2Hs/Compile/Name.hs

```diff
@@ -30,65 +32,81 @@ import Agda2Hs.Compile.Types
 import Agda2Hs.Compile.Utils
 import Agda2Hs.HsUtils
 
-isSpecialName :: QName -> Maybe (Hs.QName (), Maybe Import)
+isSpecialCon :: QName -> Maybe (Hs.QName ())
+isSpecialCon = prettyShow >>> \case
+    "Agda.Builtin.List.List"     -> special Hs.ListCon
```

Comment:
Why is List here while other type constructors are in isSpecialName?

Reply:
This separately deals with all "special" constructors in syntax (c.f. the [haskell-src-ext docs](https://hackage.haskell.org/package/haskell-src-exts-1.23.1/docs/Language-Haskell-Exts-Syntax.html#t:SpecialCon)]).